remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java

```diff
@@ -150,17 +133,31 @@ private void processSerializationContextInitializer(Collection<SerializationCont
       }
    }
 
-   private void registerProtobufMetadataManagerMBean(ProtobufMetadataManagerImpl protobufMetadataManager, GlobalComponentRegistry gcr) {
-      GlobalConfiguration globalConfig = gcr.getGlobalConfiguration();
+   @Override
+   public void cacheManagerStarted(GlobalComponentRegistry gcr) {
+      BasicComponentRegistry bcr = gcr.getComponent(BasicComponentRegistry.class);
+
+      ProtobufMetadataManagerImpl protobufMetadataManager =
+            (ProtobufMetadataManagerImpl) bcr.getComponent(ProtobufMetadataManager.class).running();
+
+      protobufMetadataManager.startInternalCache();
```

Comment:
If `ProtobufMetadataManagerImpl` needs the cache to be running, then it should start it in its own `@Start` method. 

`cacheManagerStarted` is too late: if a cache manager is created with `start=false` and 2 threads call `getCache(X)` and `getCache(Y)`, only one of them (say the one starting cache X) will wait for `cacheManagerStarted` to run. The other thread will start cache Y without waiting for the protobuf metadata cache to start first.

Reply:
Doing it on ```@Start```  leads to a dependency cycle. The error message ```Dependency cycle detected, please use ComponentRef<T> to break the cycle in path org.infinispan.factories.GlobalComponentRegistry$ModuleInitializer (a org.infinispan.factories.GlobalComponentRegistry$ModuleInitializer)``` does not contain the dependency path, only the start and the end, which are identical since it's a cycle, so not very useful. If you could do something the show the whole path would be awesome @danberindei

Reply:
@anistor that error message actually is a path of just one element, both the component name (usually the interface name) and the implementation class name are logged for each component. Let me try to run the tests myself @anistor

Reply:
Ok, so there are multiple cycles: 

* `LifecycleManager.cacheManagerStarting()` calls `running()` on the `ProtobufMetadataManagerImpl` component, but in order to start the component it needs to start the schema cache, and caches cannot start until all the modules have finished initializing (i.e. `cacheManagerStarting()`).
* A running ProtobufMetadataManagerImpl is required to start the schema cache, because `LifecycleManager.cacheStarting()` calls `running()`.

I have tentative fix here:

https://github.com/infinispan/infinispan/commit/f5d3d13c098a43c4567bea9570251c369cbf7363